src/static_data/get_static_data.go

```diff
@@ -1,217 +1,176 @@
 package main
 
 import (
+	"BGPAlert/config"
 	"fmt"
+	"golang.org/x/net/html"
 	"io"
+	"log"
 	"net/http"
 	"os"
 	"path/filepath"
+	"strings"
 )
 
 type Folder struct {
 	FolderName string
 	Urls       []string
 }
 
-func downloadFolder(folder Folder) error {
-	// Create the folder on the file system
-	err := os.MkdirAll(folder.FolderName, os.ModePerm)
+func main() {
+	// Specify the target website URL
+	//to get the bgp updates the urls are of this format; sorted by the year and month
+	//"http://routeviews.org/route-views.ny/bgpdata/2021.11/UPDATES/"
+	//the links to .bz2 files contained within the main link in the configuration file; if the links can not be attained from the main link no files will be downloaded
+	configStruct, err := config.LoadConfig("config.json")
 	if err != nil {
-		return err
+		log.Fatal("Error loading configuration:", err)
 	}
+	config.ValidDateConfiguration(configStruct)
+	mainUrl := configStruct.URLStaticData
+	fmt.Println(mainUrl)
 
-	// Download files to the folder
-	for _, url := range folder.Urls {
-		err := downloadFile(url, folder.FolderName)
+	// Make an HTTP GET request to the website
+	resp, err := http.Get(mainUrl)
+	if err != nil {
+		fmt.Printf("Error making HTTP request: %v\n", err)
+		os.Exit(1)
+	}
+	defer resp.Body.Close()
+
+	// Parse the HTML content of the response
+	doc, err := html.Parse(resp.Body)
+	if err != nil {
+		fmt.Printf("Error parsing HTML: %v\n", err)
+		os.Exit(1)
+	}
+
+	numFiles := configStruct.WindowSize
+	partialUrls := extractBZ2URLs(doc)
+	allFolders := createFolders(partialUrls, mainUrl, numFiles)
+	for _, folder := range allFolders {
+		err := downloadFolder(folder)
 		if err != nil {
-			return err
+			return
 		}
 	}
+	/*outFile := fmt.Sprintf("static_data/rawBGPData/bgpTest%d.txt", 0)
```

Comment:
are you expecting to use this commented section in the future, or can you remove it?

Reply:
using it in the next pull request for generating testing data